packages/mask/background/services/helper/request-permission.ts

```diff
@@ -1,47 +1,33 @@
 import type { Permissions } from 'webextension-polyfill'
+import { waitUntil } from '@dimensiondev/holoflows-kit'
 import { MaskMessages } from '@masknet/shared-base'
 import { getPermissionRequestURL } from '../../../shared/definitions/routes.js'
-import type { SiteAdaptor } from '../../../shared/site-adaptors/types.js'
 
-export async function requestExtensionPermission(permission: Permissions.Permissions): Promise<boolean> {
+export async function requestExtensionPermissionFromContentScript(
+    permission: Permissions.Permissions,
+): Promise<boolean> {
     if (await browser.permissions.contains(permission)) return true
-    try {
-        return await browser.permissions.request(permission).then(sendNotification)
-    } catch {
-        // which means we're on Firefox or Manifest V3.
-        // Chrome Manifest v2 allows permission request from the background.
-    }
     const popup = await browser.windows.create({
         height: 600,
         width: 400,
         type: 'popup',
         url: getPermissionRequestURL(permission),
     })
-    return new Promise((resolve) => {
+    const promise = new Promise<boolean>((resolve) => {
         browser.windows.onRemoved.addListener(function listener(windowID: number) {
             if (windowID !== popup.id) return
             browser.permissions.contains(permission).then(sendNotification).then(resolve)
             browser.windows.onRemoved.removeListener(listener)
         })
     })
+    waitUntil(promise)
```

Comment:
Do you mean?

```suggestion
    await waitUntil(promise)
```

Reply:
waitUntil is keeping the extension alive for a short time period, don't need to wait it